Framework/Algorithms/test/XrayAbsorptionCorrectionTest.h

```diff
@@ -0,0 +1,171 @@
+
+public:
+  static XrayAbsorptionCorrectionTest *createSuite() {
+    return new XrayAbsorptionCorrectionTest();
+  }
+  static void destroySuite(XrayAbsorptionCorrectionTest *suite) {
+    delete suite;
+  }
+  void test_CalculateDetectorPos() {
+    TestableXrayAbsorptionCorrection alg;
+    Kernel::V3D pos = alg.calculateDetectorPos(45, 10);
+    Kernel::V3D correctPos = {0.1, 0.1, 0};
+
+    TS_ASSERT_DELTA(pos[0], correctPos[0], 0.001);
+    TS_ASSERT_DELTA(pos[1], correctPos[1], 0.001);
+    TS_ASSERT_DELTA(pos[2], correctPos[2], 0.001);
+  }
+  void test_CalculateMuonPos() {
+    TestableXrayAbsorptionCorrection alg;
+    API::MatrixWorkspace_sptr muonProfile = createWorkspace(1);
+    API::MatrixWorkspace_sptr inputWS = createWorkspaceWithDummyShape(20);
+    std::vector<Kernel::V3D> muonPos =
+        alg.calculateMuonPos(muonProfile, inputWS, 100);
+    double y = 1.0;
+    for (auto x : muonPos) {
+      y -= 0.01;
+      TS_ASSERT_DELTA(x[0], 0, 1e-6);
+      TS_ASSERT_DELTA(x[1], y, 1e-6);
+      TS_ASSERT_DELTA(x[2], 0, 1e-6);
+    }
+  }
+  void test_NormaliseMounIntensity() {
+    TestableXrayAbsorptionCorrection alg;
+    API::MatrixWorkspace_sptr muonProfile = createWorkspace(1);
+    std::vector<double> normalisedIntensity =
+        alg.normaliseMuonIntensity(muonProfile->readY(0));
+    for (auto intensity : normalisedIntensity) {
+      TS_ASSERT_DELTA(intensity, 0.1, 1e-6);
+    }
+  }
+
+  void test_exec_with_no_shape() {
+    API::MatrixWorkspace_sptr muonProfile = createWorkspace(1);
+    API::MatrixWorkspace_sptr inputWS = createWorkspace(1);
+    Algorithms::XrayAbsorptionCorrection algo;
+    algo.initialize();
+    algo.setProperty("InputWorkspace", inputWS);
+    algo.setProperty("MuonImplantationProfile", muonProfile);
+    algo.setProperty("OutputWorkspace", "outputWS");
+    algo.setProperty("DetectorDistance", 10.0);
+    algo.setProperty("DetectorAngle", 45.0);
+    TS_ASSERT_THROWS(algo.execute(), const std::runtime_error &);
+  }
+
+  void test_exec_with_valid_shape() {
+    API::MatrixWorkspace_sptr muonProfile = createWorkspace(100);
+    auto &muonDepth = muonProfile->mutableX(0);
+    for (size_t i = 0; i < muonDepth.size(); i++) {
+      muonDepth[i] = 100;
+    }
+    API::MatrixWorkspace_sptr inputWS = createWorkspaceWithDummyShape(20);
+    Algorithms::XrayAbsorptionCorrection algo;
+    algo.initialize();
+    algo.setProperty("InputWorkspace", inputWS);
+    algo.setProperty("MuonImplantationProfile", muonProfile);
+    algo.setProperty("OutputWorkspace", "outputWS");
+    algo.setProperty("DetectorDistance", 1000.0);
+    algo.setProperty("DetectorAngle", 45.0);
+    algo.execute();
+    API::MatrixWorkspace_sptr outputWS = algo.getProperty("OutputWorkspace");
+    auto &yData = inputWS->mutableY(0);
+    for (size_t i = 0; i < yData.size(); i++) {
+      yData[i] = std::exp(-1);
+    }
+    bool result = compareWorkspace(inputWS, "outputWS", 1e-5);
```

Comment:
```suggestion
    bool result = compareWorkspace(inputWS, outputWS, 1e-5);
```

instead of overwriting the existing input, I would suggest either uploading reference data to compare with or creating a new workspace. At first glance it looks like you expect the algorithm to leave the input unchanged